src/Components/PowerBIWidget/Internal/PowerBIWidgetBuilder/PowerBIWidgetBuilder.stories.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react';
+import { ComponentStory } from '@storybook/react';
+import { getDefaultStoryDecorator } from '../../../../Models/Services/StoryUtilities';
+import PowerBIWidgetBuilder from './PowerBIWidgetBuilder';
+import { IPowerBIWidgetBuilderProps } from './PowerBIWidgetBuilder.types';
+import MockAdapter from '../../../../Adapters/MockAdapter';
+
+const wrapperStyle = { width: '100%', height: '600px', padding: 8 };
+
+export default {
+    title: 'Components/PowerBIWidgetBuilder',
+    component: PowerBIWidgetBuilder,
+    decorators: [
+        getDefaultStoryDecorator<IPowerBIWidgetBuilderProps>(wrapperStyle)
+    ]
+};
+
+type PowerBIWidgetBuilderStory = ComponentStory<typeof PowerBIWidgetBuilder>;
+
+const defaultState = {
+    id: 'mywidget',
+    type: 'PowerBI',
+    widgetConfiguration: {
+        type: 'tile',
+        label: '',
+        reportId: ''
+    }
+};
+
+const Template: PowerBIWidgetBuilderStory = (args) => {
+    const [formData, updateWidgetData] = useState(
+        args.formData || defaultState
+    );
+    return (
+        <>
+            <PowerBIWidgetBuilder
+                formData={formData}
+                updateWidgetData={updateWidgetData}
+                {...args}
+            />
+        </>
+    );
+};
+
+export const WithMockAdapter = Template.bind(
```

Comment:
do we want some stories with different types of widgets? or field selections? I think you were going to make some fields how or hide based on different selections, can we automate those with interaction tests?

Reply:
We only show/hide based on type being Report/Dashboard or Visual/TIle. Currently Visual/TIle are the only values we are going to support so we can't get into a state where the logic would differ.